indra/newview/llagentcamera.cpp

```diff
@@ -941,28 +941,19 @@ void LLAgentCamera::cameraZoomIn(const F32 fraction)
 		}
 	}
 
-	new_distance = llmax(new_distance, min_zoom); 
+	new_distance = llmax(new_distance, min_zoom);
 
-	F32 max_distance = getCameraMaxZoomDistance();
-
-    max_distance = llmin(max_distance, current_distance * 4.f); //Scaled max relative to current distance.  MAINT-3154
-
-	if (new_distance > max_distance)
+	// Unless camera is unlocked
+	if (!gSavedSettings.getBOOL("DisableCameraConstraints"))
```

Comment:
How frequent is the cameraZoomIn call? I think LLCachedControl variant should be used here.